include/image.h

```diff
@@ -125,22 +157,42 @@ class Image
             + ")";
     }
 
+    const bool operator == (const Image &other) const {
```

Comment:
Does this leading `const` do anything?  (I don't think so, but it's a question because I'm not absolutely positive).